[CDAP-18557] Trim down runrecord when runtime service caches it in lo…

wyzhang opened the pull request:
…cal storage.

Why:
RunRecord could be quite large due to containing large system args (i.e.
containing plugin documentations when pipeline runs are triggered based on events)
This leads to large-sized value stored in local levelDB by runtime service.
The large-sized value could lead to disk overhead and out of memory problems
for runtime service. As runtime record is used only for tracking inflight
program runs to perform validations, there is no need to store entire
runrecord.

runtime

chtyim commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/DirectRuntimeRequestValidator.java:
```diff
@@ -131,7 +132,10 @@ private void insertRunRecord(ProgramRunId programRunId, RunRecordDetail runRecor
+        // Strip off user and system args as runtime only needs the run status for validation purpose.
+        // User and system args could be large and store them in local store can lead to unnecessary storage
+        // and processing overhead.
+        store.recordProgramProvisioning(programRunId, Collections.EMPTY_MAP, Collections.EMPTY_MAP,
```
Use Collections.emptyMap() instead

wyzhang replied:
Fixed.

chtyim reviewed (approved):
Just one comment about using the method instead of constant for empty map

chtyim commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/DirectRuntimeRequestValidator.java:
```diff
@@ -159,4 +165,15 @@ private void insertRunRecord(ProgramRunId programRunId, RunRecordDetail runRecor
+
+  private Map<String, String> trimSystemArgs(Map<String, String> args) {
+    Map<String, String> trimmed = new HashMap<>();
+  if (args.containsKey(SystemArguments.PROFILE_NAME)) {
```
indentation

wyzhang replied:
The file you reviewed was out of date. Latest change is fine

wyzhang committed:
[CDAP-18557] Trim down runrecord when runtime service caches it in local storage.

Why:
RunRecord could be quite large due to containing large system args (i.e.
containing plugin documentations when pipeline runs are triggered based on events)
This leads to large-sized value stored in local levelDB by runtime service.
The large-sized value could lead to disk overhead and out of memory problems
for runtime service. As runtime record is used only for tracking inflight
program runs to perform validations, there is no need to store entire
runrecord.
